### DISCUSSION TITLE
Filter out contaminants - forest.py

### DISCUSSION THREAD

**Discussion by Edouard94** [TheBrownLab/PhyloFisher#93] [Q&A] [answered]
Hello everyone,

I am contacting you to seek some help at the tree parsing step of PhyloFisher workflow,

The data I am using is from data mining of NCBI omic data to look for potential parasite infection in arthropod hosts. I have run forest.py and obtained SGTs for each gene/protein that was recovered during the data mining (blast hit). Those proteins are a mix of specific and conserved proteins of my parasites and can happen to be shared with the host. So, the manual parsing will take me some time, as I have around 150 SGTs.

Then I wonder if I could use the _--contaminants <contams.tsv>_ option to filter out automatically host contaminant. But I am not sure it would work as some host sequence/branch are clustering with the parasite proteome, if that makes sense?

Here an example of one of my SGT (you can see the separation between the microsporidia and arthropod clade):

![image](https://github.com/TheBrownLab/PhyloFisher/assets/60606060/e3671c6a-4467-4f12-bcf1-8d057b17f498)

Then, I am the wondering if I could use the tsv file to remove the contaminants:
Microsporidia	Insecta	        HigherTaxonomy
Microsporidia	Tardigrada	HigherTaxonomy
Microsporidia	Chelicerata	HigherTaxonomy
Microsporidia	Myriapoda	HigherTaxonomy
Microsporidia	Crustacea	HigherTaxonomy

But would that remove the host sequence 'parasitised' as well? I am also not sure of the UniqueID.

Thank you for your insights on this!

**Comment by atice** [comment]
Hi @Edouard94 

Apologies for my delayed reply. I am transitioning to a faculty position and the university has been keeping me quite busy. 

I will definitely try and help you as much as I can but I may need some clarification to do so. Is the incomplete image of a parasorter display what you meant to send and not the entire thing? I only see one microsporidian taxon Metchnikovella incurvata.

I don't quite understand what you mean by:

"But would that remove the host sequence 'parasitised' as well? I am also not sure of the UniqueID."

Is the dilemma that sometimes clades of parasite sequences and clades or insect sequences are "real" and other times they represent contamination? 

Just FYI. If you choose to use the --contaminants or --backpropogate all the script does is mark "d" for delete in the tsv files that are displayed by parasorter. So if you do not like the results you can simply change the decision back to "o" or "p" if you wish and that is what the sequences will be designated as when you run apply_to_db.py.

Let me know how I can clarify or help further.

Thank you for considering PhyloFisher for your research.

Alex Tice

**Comment by Edouard94** [comment]
Hello Alex,

No worries I totally understand and thank you a lot for coming back to me. 

Yes, the image below is an incomplete SGT mostly showing a 'microsporidia clade' (so what I want to keep) and the start of an 'arthropod clade' where I manually put the sequences to be deleted (tell me if you would need a full example).

I think you got my dilemma: I want to keep the 'contaminated' host sequences that could actually correspond to a microsporidian parasite. Where putative contaminations (i.e. hosts parasitised) are clustering with the microsporidia reference data, such as shown on the image (cluster around Metchnikovella reference).
So, I would like to only keep the 'contaminated' host sequences, where I suppose that microsporidia were sequenced and preserved after the host genome assembly. I already have assumptions that it is the case, after 'mining' the WGS data with blast and a set of microsporidia proteins.

I am then just worried that the _-- contaminants_ option would also mark the interesting/contaminated hosts sequences to be deleted. I can try it and see what I get.
Was the example I gave all right as a .tsv file and the uniqueID correct in my first post?

My best wishes for your new position and thank you for your help,
Edouard

**Reply by atice** [reply]
Hi @Edouard94,

If you don't mind providing me one full example with the decisions marked as you fell is correct that would help me decide if there is an existing automated way to get the same result, if there is a quick edit to the code we could make to give you one, or if for now you will have to examine all 150 by eye.

@robert-ervin-jones is looking into wether or not on UniqueIDs are accepted as the first column in the .tsv provided to the --contaminants flag or if all levels of PhyloFisher taxonomy are accepted. So we will get back to you on that aspect soon.

Cool project by the way! Hope we can help.

Alex

**Reply by Edouard94** [reply]
Hi Alex,

Thank you for your response. I attach here a WeTransfer with a full example of a parsed SGT (I used WeTransfer as the files' formats are not supported by GitHub): https://we.tl/t-FKa4IWOSTC

So, you will normally see that the only clade we want to delete is the 'arthropod' one, when we want to keep the microsporidian + fungal ones.

Let me know if there is a way to do it automatically, I am doing it manually for now.

Thank you for your interest, we will publish this work, hopefully soon!

Best wishes,
Edouard

**Comment by atice** [comment, marked as answer]
Hi Edouard,

After a lot of careful consideration, we can't come up with an immediate automated solution to your issue. There is probably potential in adding functionality that would take a group of interest and say "if more member of that group are in this clade than any other keep all sequences. All other clades with less of that group mark some other taxa for deletion." However, with me getting settled in at my new university and role as a laboratory head and Robert beginning his dissertation writing we don't have the time between us right now to implement and test this kind of additional functionality.

As screening genomes for interesting contaminants was something we noticed as a possibility when building the PhyloFisher database, we will consider adding fucntionality to ease the burden of this process when we begin to move forward with v2.

I hope the manual parsing goes as smoothly as possible. I apologise we can't help in this instance. If you think of some creative way to automate something like this don't hesitate to let us know. PhyloFisher was always meant to be a tool for the community, and we are always looking for ways to better serve that community.

Let us know if you have any further questions.

Alex

Alexander K. Tice, Ph.D.
Department of Biological Sciences
Mississippi State University
027 Harned Hall
Lab Telephone: 662-325-4008

________________________________

**Reply by Edouard94** [reply]
Dear Alex,

Thank you for your response, I am happy that our discussion can bring up new ideas for the development of PhyloFisher.

The if function you propose sounds like a good option for this particular tree parsing. Let see if we could implement this in the `forest.py` script, but for now the manual parsing is going smoothly yes :)

Thank you for your time and my best wishes,
Edouard
